libguard/guard_interface.cpp

```diff
@@ -166,6 +166,16 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
                 lastPos++;
                 continue;
             }
+            else if ((existGuard.errType == GARD_User_Manual) &&
```

Comment:
- But, we need to allow the requested hardware to guard if that
  particular hardware was guarded by the user because that hardware
  might be requested to guard due to some hardware failures and
  we may need to analyze failures by using the associated error log.

Little confused with use case.  Manual guard creates guard record right?  Aree you trying to solve the run time manual guard follwed by actual hardware failure use case?

Please include tested details.

Reply:
> * But, we need to allow the requested hardware to guard if that
>   particular hardware was guarded by the user because that hardware
>   might be requested to guard due to some hardware failures and
>   we may need to analyze failures by using the associated error log.
> 
> Little confused with use case. Manual guard creates guard record right? Aree you trying to solve the run time manual guard follwed by actual hardware failure use case?
> 
> Please include tested details.

We won't support runtime manual guard operation.

Here, I am trying to convey "If we already have manual guard record for the specific target and the later system attempts to create Fatal/Predictive guard record for the same resource then, we said it already exists but, I think we should allow that since that might happen because of some error and we need to fix that by looking at the error but, that's not possible now since we are not allowing to override the manual guard record by the Fatal/Predictive.

Reply:
> > * But, we need to allow the requested hardware to guard if that
> >   particular hardware was guarded by the user because that hardware
> >   might be requested to guard due to some hardware failures and
> >   we may need to analyze failures by using the associated error log.
> > 
> > Little confused with use case. Manual guard creates guard record right? Aree you trying to solve the run time manual guard follwed by actual hardware failure use case?
> > Please include tested details.
> 
> We won't support runtime manual guard operation.
> 
> Here, I am trying to convey "If we already have manual guard record for the specific target and the later system attempts to create Fatal/Predictive guard record for the same resource then, we said it already exists but, I think we should allow that since that might happen **because of some error and we need to fix that** by looking at the error but, that's not possible now since we are not allowing to override the manual guard record by the Fatal/Predictive.

If the manual guard is created only at bmc ready state , BMC boot code will remove the guarded units from functional list , Are you trying to handle the special cases , which ignore guard records apply with special rules? If that case need beter description in commit message.

Reply:
> > * But, we need to allow the requested hardware to guard if that
> >   particular hardware was guarded by the user because that hardware
> >   might be requested to guard due to some hardware failures and
> >   we may need to analyze failures by using the associated error log.
> > 
> > Little confused with use case. Manual guard creates guard record right? Aree you trying to solve the run time manual guard follwed by actual hardware failure use case?
> > Please include tested details.
> 
> We won't support runtime manual guard operation.
@RameshIyyar we can create guard records using guard tool any time right?
> 
> Here, I am trying to convey "If we already have manual guard record for the specific target and the later system attempts to create Fatal/Predictive guard record for the same resource then, we said it already exists but, I think we should allow that since that might happen **because of some error and we need to fix that** by looking at the error but, that's not possible now since we are not allowing to override the manual guard record by the Fatal/Predictive.

Reply:
>If the manual guard is created only at bmc ready state , BMC boot code will remove the guarded units from functional list 

Do you mean, if it is not functional then, further FWs stack BMC (post UpdateHwModel, NoOp before this istep in the BMC), SBE, HB, and PHYP won't touch that resource and create the error log with Fatal/Predictive guard request if there is an error on that resource? 

> we can create guard records using guard tool any time right?

Sorry, I meant from the customer interface.

Reply:
I have added details in the last commit message like, this case can occur in the resource recovery.